packages/client/src/components/uppy/uppy.ts

```diff
@@ -0,0 +1,257 @@
+import Uppy, { UppyFile } from '@uppy/core'
+import AwsS3Multipart, { AwsS3Part } from '@uppy/aws-s3-multipart'
+import {
+  AbortMultipartUploadDocument,
+  AbortMultipartUploadMutationVariables,
+  CompleteMultipartUploadDocument,
+  CompleteMultipartUploadMutationVariables,
+  CompleteMultipartUploadReturn,
+  PrepareUploadPartsDocument,
+  PrepareUploadPartsQueryVariables,
+  GetUrlReturn,
+  CreateMultipartUploadDocument,
+  CreateMultipartUploadReturn,
+  ListPartsDocument,
+  ListPartsQueryVariables,
+  useDownloadGetUrlQuery,
+  useDownloadGetUrlsQuery,
+  GetUrlsReturn,
+} from './graphql'
+import { createRequest } from 'urql'
+import type { Client } from 'urql'
+
+const uploadedParts: { [x: string]: any[] } = {}
+
+function getUploadedParts(uploadId: string) {
+  const _uploadedParts = uploadedParts[uploadId] || []
+  uploadedParts[uploadId] = _uploadedParts
```

Comment:
memory leak

Reply:
uploadedParts[uploadId] is now deleted on upload completition or abortion